### PR TITLE
Fix flake8 for main files

### PR DESCRIPTION
**Pull request by lucasmoura**
This MR fixes flake8 warnings for the main files of the project. It also:

* Excludes flake8 checking the test files, as stated in #386 
* Add flake8 checking in .travis.yml, but now it only look for the main python files in the project

This MR also adds a TODO in the text.py code related to #398 and is related also to #304

### CODE REVIEW

**Comment by kaivalyar** [comment]
I think line 19 adds flake8 testing to the travis build process again.

I am not sure if we should add this again, after it was explicitly removed in #386. While style compliance is important, I am not sure if it is necessary to have travis use flake8 to enforce it for the builds to pass.

**Comment by lucasmoura** [comment]
@kaivalyar I think for the test files, it is true that flake8 was really not that usefull, since the intended users of the code would not actually read the test files. But for the main files, I do believe it is important.
It makes easier to read code, while also making it easier to contribute and review code.

Also, issues such #398 and the #337 would not be there if we allow flake8 analysis to break the build. Besides that, any unecessary check for the project can be added to the flake8 config file.

**Comment by kaivalyar** [comment]
@lucasmoura yes, I agree. Style is definitely important for readability. How strictly should it be enforced, however, is something subjective we should discuss further. What to exclude using the flake config file, for example.

**Comment by norvig** [comment]
I think there is no need for flake8 on the test files.

If tyhere is bad style, we can catch examples of it individually, without imposing automated tests,

File: agents.py [hunk 1]
```diff
@@ -394,8 +395,9 @@ def things_near(self, location, radius=None):
         if radius is None:
             radius = self.perceptible_distance
         radius2 = radius * radius
-        return [(thing, radius2 - distance_squared(location, thing.location)) for thing in self.things
```
Comment (norvig) [inline agents.py]: I actually like this indentation better than the correction

**Comment by kaivalyar** [comment]
@norvig what about other non-test files though, should we run flake8 upon them for testing pep8 compatibility before sending in future PRs?

**Pull request merged**